test/blackbox-tests/test-cases/lib-collision/lib-collision-public-same-folder.t

```diff
@@ -19,9 +19,9 @@ the same folder.
 Without any consumers of the libraries
 
   $ dune build
-  Error: Library foo is defined twice:
-  - dune:6
-  - dune:3
+  Error: Multiple rules generated for _build/default/foo.cmxs:
```

Comment:
Attempt to fix in https://github.com/anmonteiro/dune/pull/10 (still requires the introduction of `Library.private_name`, I am not able to see a way around this).

Reply:
I think we'll want @rgrinberg to review those changes. I don't know exactly what semantics we want here.

Reply:
I think the old error was good. The user clearly made a mistake here.

Reply:
what do you think about comparing by obj_dir instead of exposing the private name (https://github.com/anmonteiro/dune/pull/10#discussion_r1540119767)? I guess that wouldn't cover private names in different directories.

Reply:
https://github.com/anmonteiro/dune/pull/10/files#r1540920892